crates/rust-eigenda-v2-client/src/cert_verifier.rs

```diff
@@ -82,25 +63,44 @@ impl<S> CertVerifier<S> {
         Ok(quorums.to_vec())
     }
 
-    /// Calls the VerifyCertV2 view function on the EigenDACertVerifier contract.
+    /// Calls the CheckDACert view function on the EigenDACertVerifier contract.
     ///
     /// This method returns an empty Result if the cert is successfully verified. Otherwise, it returns a [`CertVerifierError`].
-    pub async fn verify_cert_v2(&self, eigenda_cert: &EigenDACert) -> Result<(), CertVerifierError>
+    pub async fn check_da_cert(&self, eigenda_cert: &EigenDACert) -> Result<(), CertVerifierError>
     where
         EthersSigner<S>: Signer,
     {
-        self.cert_verifier_contract
-            .verify_da_cert_v2(
-                eigenda_cert.batch_header.clone().into(),
-                eigenda_cert.blob_inclusion_info.clone().into(),
-                eigenda_cert.non_signer_stakes_and_signature.clone().into(),
-                eigenda_cert.signed_quorum_numbers.clone().into(),
-            )
+        let abi_encoded_cert: Vec<u8> = eigenda_cert_to_abi_encoded(eigenda_cert)?;
+        let res = self
+            .cert_verifier_contract_base
+            .check_da_cert(Bytes::from(abi_encoded_cert))
             .call()
             .await
-            .map_err(|_| CertVerifierError::Contract("verify_cert_v2".to_string()))?;
+            .map_err(|_| CertVerifierError::Contract("check_da_cert".to_string()))?;
+        if res != CHECK_DA_CERT_STATUS_SUCCESS {
+            return Err(CertVerifierError::VerificationFailed(
+                "check_da_cert returned non-succesfull value".to_string(),
```

Comment:
please encode the actual status code in the structured error, as that will be very useful information to the client.
Also I feel like the first error (forget what its called, NULL_TYPE or something) should be treated very differently, as its NOT actually a cert error. prob want to panic and raise the alarms that the contract is buggy if that happens.